xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/suggest/suggest.js

```diff
@@ -730,7 +730,10 @@ var XWiki = (function(XWiki){
         noHighlight: true, // we do the highlighting ourselves
         containerTagName: 'a'
       });
-      item.containerElement.setAttribute('href', arr[i].url);
+      if (arr[i].url == null)  {
+        item.containerElement.setAttribute('href', 'javascript:void(0)');
+      }
+      else item.containerElement.setAttribute('href', arr[i].url);
```

Comment:
Strange formatting. Also, you should always wrap if/else blocks in ``{...}``. But in this case you can simplify it to:

```suggestion
      item.containerElement.setAttribute('href', arr[i].url || 'javascript:void(0)');
```